ambry-store/src/test/java/com.github.ambry.store/IndexTest.java

```diff
@@ -872,7 +1079,10 @@ private MockId getUniqueId() {
    * @throws IOException
    */
   private void appendToLog(long size) throws IOException {
-    ByteBuffer buffer = ByteBuffer.allocate((int) size);
+    byte[] bytes = TestUtils.getRandomBytes((int) size);
+    // ensure atleast one byte is set to 1
+    bytes[random.nextInt((int) size)] = 1;
```

Comment:
what if this byte is before/after the hard delete range inside the blob? Hard delete starts at a non zero index at the blob and ends before the end of the blob. Maybe you need to make sure it is written inside the range that will be hard deleted.